### ISSUE TITLE
How to handle new page from "target=_blank" event?

### ISSUE THREAD

**Issue by shaoerkuai** [MarketSquare/robotframework-browser#3925]
I don't know the keyword to use. In pure Python I could use `with context.expect_page()`, but it seems no keyword like this already existed..

How could I make it happen in robot-browser?

**Comment by zhaoyuhong5** [comment]
Here is a brief explanation on how to handle the new page in a "target=_blank" event:

### Understanding "target=_blank"
When an HTML link has the attribute "target=_blank", clicking on it will open the linked page in a new browser tab or window. This is commonly used to direct users to external resources or other relevant pages while keeping the original page accessible.

### Handling in JavaScript (Common Approach)

#### 1. Using the `window.open` Event and Managing the New Window/Tab
```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <title>Target Blank Handling</title>
    <script>
        function handleNewPage() {
            const newWindow = window.open();  // Opens a new window or tab
            // You can now perform operations on this new window
            // For example, set its size, position, etc.
            newWindow.resizeTo(800, 600);
            newWindow.moveTo(100, 100);
            // Or listen for events on the new window like its load event
            newWindow.addEventListener('load', function () {
                console.log('The new page has loaded');
            });
        }
    </script>
</head>

<body>
    <a href="https://example.com" target="_blank" onclick="handleNewPage(); return false;">Click me</a>
</body>

</html>
```
Explanation:
- The `window.open` function is called when the link is clicked (through the `onclick` event handler). It creates the new window or tab.
- We can then customize the new window's properties like its size (`resizeTo`) and position (`moveTo`).
- Also, we can attach event listeners to the new window. In the example, we listen for the `load` event to know when the new page has fully loaded. Note that returning `false` in the `onclick` event handler prevents the default behavior of the link from immediately opening the page, allowing us to manage it first.

#### 2. Security and Cross-Origin Considerations
When working with the new window created by `window.open`, if the new page is from a different origin (domain, protocol, or port), there are security restrictions due to the browser's same-origin policy. For example, you won't be able to directly access the DOM elements of the new page from the original page's JavaScript code if they are from different origins.
To communicate between pages from different origins in a limited way, you can use techniques like the `postMessage` API. The originating page can send a message like this:
```javascript
newWindow.postMessage('Some data', 'https://example.com');
```
And the receiving page (in the new window) can listen for the message:
```javascript
window.addEventListener('postMessage', function (event) {
    if (event.origin === 'https://your-origin.com') {
        console.log('Received message:', event.data);
    }
});
```

### Handling in Other Programming Languages (Server-Side)
If you want to handle the new page from a server-side perspective (for example, in a web application using Python with a framework like Flask or Django), you can track the clicks on links with "target=_blank" and perform actions such as logging, analytics, or even modifying the page content that is about to be served in the new window based on certain conditions. However, the specific implementation would depend on the server-side framework and its capabilities.

In summary, handling the new page in a "target=_blank" event mainly involves using JavaScript to manage the new window/tab properties and dealing with cross-origin communication if necessary, while on the server side, it can be about tracking and potentially influencing the page content in the new window based on application requirements.

**Comment by aaltat** [comment]
I don’t quite understand your question but ‘Switch Page    NEW` should work. If it doesn’t, then could you explain what else you have tried and what errors you get and what actually happened?
